### DISCUSSION TITLE
How to mask regions of image?

### DISCUSSION THREAD

**Discussion by ruckc** [ultralytics/ultralytics#762] [Q&A]
I have an video feed and I want to mask out detections in various regions with an image mask.  I've got the image masked with the irrelevant sections as black, but yolo keeps detecting things around the edges of it.  Is there anyway to have yolo completely ignore pure black regions or should I use an alpha channel?  Just trying to reduce false positives (i.e. skateboards in the clouds).

**Comment by glenn-jocher** [comment]
@ruckc hello! To have YOLOv8 ignore certain regions of an image, you can modify the dataset to include negative examples or use preprocessing to mask out the regions before passing the images to the model.

For the masking approach, you don't need to use an alpha channel; instead, ensure that the masked regions are set to a constant value that the model can learn to ignore. If you're still getting detections around the edges, consider applying a blur or smoothing the edges of the mask to reduce edge artifacts that might be causing false detections.

Additionally, you can augment your training data with examples of the regions you want to ignore, specifically labeled as negative examples, so the model learns to not detect objects in those areas.

For more detailed guidance on training and preprocessing, you can refer to our documentation on the Train mode at https://docs.ultralytics.com/modes/train. Keep in mind that careful dataset preparation and model tuning are key to reducing false positives. 😊🚀
